### PR TITLE
10 add a user account page

### PR DESCRIPTION
**Pull request by shrprabh**
feat: Implement user account management with email update, OTP verification, and account deletion

- Added `UserAccount` component to manage user information, including username, email, and roles.
- Implemented the ability to update the user's email address with OTP verification:
  - User can request an OTP to be sent to the new email address.
  - User can verify the OTP to confirm the email change.
  - Added automatic logout after successful email change.
- Added a feature to resend the OTP if the user didn't receive it.
- Integrated a delete account feature with confirmation dialog:
  - User is prompted with a confirmation dialog before account deletion.
  - If confirmed, the account is deleted via an API call.
- Integrated MUI components (`Card`, `Typography`, `TextField`, `Button`, etc.) for a responsive and accessible UI.
- Used `react-toastify` for providing feedback to users with success and error messages.
- Used `useAuth` and `useLogout` hooks to manage authentication and logout functionality.
- Ensured proper state management using `useState` hooks for handling inputs, modals, and confirmation visibility.

### CODE REVIEW

File: src/features/users/UserAccount.tsx [hunk 1]
```diff
@@ -1,15 +1,425 @@
...
+        currentEmail: email,
+        newEmail: values.newEmail
+      }).unwrap()
+
+      showMessageDialog('OTP is sent successfully to your new email address.')
+      dispatch({ type: 'TOGGLE_OTP_MODAL', payload: true }) // Open OTP modal
+    } catch (error) {
+      const customError = error as CustomError
+      if (customError.status === 409) {
+        showMessageDialog('Email address already exists')
```
Comment (dsclassen) [inline src/features/users/UserAccount.tsx:120]: Should be:

"This Email is already associated with another account"

File: src/features/users/UserAccount.tsx [hunk 2]
```diff
@@ -1,15 +1,425 @@
...
+      } else if (customError.status === 400) {
+        showMessageDialog(customError.data.message)
+      } else {
+        showMessageDialog('Error sending an OTP to your new email address')
+      }
+    }
+  }
+
+  const handleVerifyOtp = async (values: { otp: string }) => {
+    console.log('Verifying OTP:', values.otp)
```
Comment (dsclassen) [inline src/features/users/UserAccount.tsx:130]: Please comment out so console logs don't appear in production deployment.

File: src/features/users/UserAccount.tsx [hunk 3]
```diff
@@ -1,15 +1,425 @@
...
+  }
+
+  const logout = useLogout()
+  const handleAutomaticLogout = () => {
+    logout()
+  }
+
+  const handleUpdateEmail = async (values: { newEmail: string }) => {
+    try {
+      console.log('Updating email to:', values.newEmail)
```
Comment (dsclassen) [inline src/features/users/UserAccount.tsx:108]: Please comment out so console logs don't appear in production deployment.

File: src/features/users/UserAccount.tsx [hunk 4]
```diff
@@ -1,15 +1,425 @@
...
+  }
+
+  const confirmDeleteAccount = async () => {
+    if (!username) {
+      showMessageDialog('Username is undefined. Cannot delete account.')
+      return
+    }
+
+    try {
+      console.log('Deleting account:', username)
```
Comment (dsclassen) [inline src/features/users/UserAccount.tsx:186]: Please comment out so console logs don't appear in production deployment.

File: src/slices/usersApiSlice.ts
Comment (dsclassen) [inline src/slices/usersApiSlice.ts:1]: Please revert this change. Your feature does not need to edit this file. Keep your PR clean

**Review by dsclassen — CHANGES_REQUESTED** [review]
Fix these small items and I will merge

**Comment by dsclassen** [comment]
In the rare cases when either the new email address has a typo or for some reason can't be delivered I think it would be good to perhaps add a little "X" to the OTP entry dialog box so it can be dismissed.

File: src/features/users/UserAccount.tsx (hunk 1 above)
Comment (shrprabh) [reply src/features/users/UserAccount.tsx]: Done

File: src/features/users/UserAccount.tsx (hunk 2 above)
Comment (shrprabh) [reply src/features/users/UserAccount.tsx]: Done

File: src/features/users/UserAccount.tsx (hunk 3 above)
Comment (shrprabh) [reply src/features/users/UserAccount.tsx]: Done

File: src/slices/usersApiSlice.ts
Comment (shrprabh) [reply src/slices/usersApiSlice.ts:1]: Done

**Review by dsclassen — APPROVED** [review]

**Pull request merged**